engine/access/rest/request/height.go

```diff
@@ -0,0 +1,75 @@
+package request
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+)
+
+const sealed = "sealed"
+const final = "final"
+
+// Special height values
+const SealedHeight uint64 = math.MaxUint64 - 1
+const FinalHeight uint64 = math.MaxUint64 - 2
+const EmptyHeight uint64 = math.MaxUint64 - 3
+
+type Height uint64
+
+func (h *Height) Parse(raw string) error {
+	if raw == "" { // allow empty
+		*h = Height(EmptyHeight)
+		return nil
+	}
+
+	if raw == sealed {
+		*h = Height(SealedHeight)
+		return nil
+	}
+	if raw == final {
+		*h = Height(FinalHeight)
+		return nil
+	}
+
+	height, err := strconv.ParseUint(raw, 0, 64)
+	if err != nil {
+		return fmt.Errorf("invalid height format")
+	}
+
+	*h = Height(height)
+	return nil
+}
+
+func (h Height) Flow() uint64 {
+	return uint64(h)
+}
+
+type Heights []Height
+
+func (h *Heights) Parse(raw []string) error {
+	var height Height
+	heights := make([]Height, 0)
+	for _, r := range raw {
+		err := height.Parse(r)
+		if err != nil {
+			return err
+		}
+		// don't include empty heights
+		if height == Height(EmptyHeight) {
+			continue
+		}
+
+		heights = append(heights, height)
+	}
```

Comment:
is it worth also detecting duplicate heights?

Reply:
for sure